src/box/lua/config/utils/schema.lua

```diff
@@ -18,6 +18,65 @@ local schema_mt = {}
 
 local scalars = {}
 
+-- {{{ Array helpers
+
+-- Convert array to string
+local function arr2str(array)
+    if type(array) ~= "table" then return tostring(array) end
+    local result = "["
+    for i, v in ipairs(array) do
+        if type(v) == "table" then
+            result = result .. arr2str(v)
+        else
+            result = result .. tostring(v)
+        end
+        if i < #array then
+            result = result .. ", "
+        end
+    end
+    result = result .. "]"
+    return result
+end
+
+-- Determine with a Lua table can be treated as an array.
+-- Explicitly returns "not an array" for very sparse arrays.
+-- Returns:
+-- -1   Not an array
+-- 0    Empty table
+-- >0   Highest index in the array
+--
+--     -- {[1] = 2}       - array (!)
+--     -- {1, 2, 3}       - array
+--     -- {1, 2, nil}     - array
+--     -- {} or {nil}     - not array
+--     -- {1, 2 a = 1}    - not array
+--     -- {a = 1, b = 2}  - not array
+--
+-- Don't use validate_table_is_array because we don't need to
+-- detect any type misses here.
+--
+-- Taken from lua-cjson
+-- Copyright (c) 2010-2012  Mark Pulford <mark@kyne.au>
+local function is_array(table)
+    local max = 0
+    local count = 0
+    for k, _ in pairs(table) do
+        if type(k) == "number" then
+            if k > max then max = k end
+            count = count + 1
+        else
+            return -1
+        end
+    end
+    if max >= count * 2 then
+        return -1
+    end
+
+    return max
+end
```

Comment:
We have a function in this exact file that is called `validate_table_is_array`, I believe you should find a way to reuse it or generalize the implementation.